### PR TITLE
feat(eslint): enable jest rules

### PR DESCRIPTION
**Pull request by stevensacks**
Enable jest eslint rules. The plugin was installed but it wasn't being used.

### CODE REVIEW

**Comment by igetgames** [comment]
The jest rules are added in [test.js](https://github.com/americanexpress/eslint-config-amex/blob/master/test.js), so this patch should target that file.

**Comment by stevensacks** [comment]
Ah, I see. Ok, I'll fix this.

**Comment by stevensacks** [comment]
I just pushed a fix for a rule I noticed was conflicting with co-located tests.

File: __tests__/package.spec.js [hunk 1]
```diff
@@ -12,6 +12,8 @@
  * the License.
  */
 
+/* eslint-disable jest/no-test-callback */
```
Comment (PixnBits) [inline __tests__/package.spec.js:15]: well that's embarrassing

might be better to 
```javascript
const path = require('path');
const util = require('util');
const exec = util.promisify(require('child_process').exec);
const packageJson = require('../package.json');
describe('package.json', () => {
  it('should not have peer dep warnings', async () => {
    expect.assertions(2);
    // ex:
    // npm ERR! peer dep missing: eslint@<2.3.0, required by babel-eslint@5.0.4
    const { stdout, stderr } = exec('npm ls', { cwd: path.resolve(__dirname, '../') });
    // if there are no complaints from npm, stdout is null
    expect(stdout || '').not.toMatch('peer dep missing');
    expect(stderr || '').not.toMatch('peer dep missing');
  });
```
than disabling rules in our own repository/project?

Comment (stevensacks) [reply __tests__/package.spec.js:15]: It's not my code and I didn't immediately know how to make it compliant, so I disabled the rules as a quick solution. If you know how to fix it, please do.

Comment (stevensacks) [reply __tests__/package.spec.js]: Fixed

**Comment by PixnBits** [comment]
I think these were intended to be active but were overlooked? If that's the case, @10xLaCroixDrinker do you think this might be a fix? Adding breaking rules might be a major version feature? 🤔

**Comment by 10xLaCroixDrinker** [comment]
> I think these were intended to be active but were overlooked? If that's the case, @10xLaCroixDrinker do you think this might be a fix? Adding breaking rules might be a major version feature? 🤔

Arent we bundling all these PRs into a major release anyway?

**Pull request merged**